autorest/azure/environments.go

```diff
@@ -101,6 +113,16 @@ var (
 	}
 )
 
+// EnvironmentFromName returns an Environment based on the common name specified
+func EnvironmentFromName(name string) (Environment, error) {
+	name = strings.ToUpper(name)
+	env, ok := environments[name]
+	if !ok {
+		return env, fmt.Errorf("There is no cloud envionment matching the name '%s'!", name)
```

Comment:
hmm I wish we were starting errors like `autorest/azure:` but I realized many files in this pkg already don't do it. so it is fine.